src/coreclr/vm/peimagelayout.cpp

```diff
@@ -721,14 +729,57 @@ FlatImageLayout::FlatImageLayout(PEImage* pOwner)
         _ASSERTE((offset - mapBegin) < mapSize);
         _ASSERTE(mapSize >= (UINT64)size);
 
-        char *addr = (char*)CLRMapViewOfFile(m_FileMap, FILE_MAP_READ, mapBegin >> 32, (DWORD)mapBegin, (DWORD)mapSize);
-        if (addr == NULL)
+        LPVOID view = CLRMapViewOfFile(m_FileMap, FILE_MAP_READ, mapBegin >> 32, (DWORD)mapBegin, (DWORD)mapSize);
+        if (view == NULL)
             ThrowLastError();
 
-        addr += (offset - mapBegin);
-        m_FileView.Assign((LPVOID)addr);
+        m_FileView.Assign(view);
+        addr = (LPVOID)((size_t)view + offset - mapBegin);
+
+        INT64 uncompressedSize = pOwner->GetUncompressedSize();
+        if (uncompressedSize > 0)
+        {
+#if defined(CORECLR_EMBEDDED)
+            HandleHolder anonMap = WszCreateFileMapping(INVALID_HANDLE_VALUE, NULL, PAGE_READWRITE, uncompressedSize >> 32, (DWORD)uncompressedSize, NULL);
+            if (anonMap == NULL)
+                ThrowLastError();
+
+            LPVOID anonView = CLRMapViewOfFile(anonMap, FILE_MAP_READ | FILE_MAP_WRITE, 0, 0, 0);
+            if (anonView == NULL)
+                ThrowLastError();
```

Comment:
This is worth a comment. Normally I would expect a simple memory allocation. I assume we're using file mapping so that the rest of the code doesn't have to change, right?
Do we know if this approach has some significant drawback compared to just a simple allocation?

Reply:
I have added a detailed comment here.